test/integration_test.go

```diff
@@ -360,6 +360,14 @@ func (ts *IntegrationTestSuite) TestCascadingCancellation() {
 	ts.NotNil(run)
 	ts.NoError(err)
 
+	// Need to give workflow time to start its child
+	for {
+		_, err := ts.client.DescribeWorkflowExecution(ctx, childWorkflowID, "")
+		if err == nil {
+			break
+		}
```

Comment:
Is it possible that this would keep the for loop for ever?

Reply:
Yeah, I'll select against a timeout and fail the test in case